apiserver/services/controllerservices.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package services
+
+import (
+	"context"
+
+	"github.com/juju/juju/controller"
+	"github.com/juju/juju/core/crossmodel"
+	"github.com/juju/juju/core/watcher"
+	modelmanagerservice "github.com/juju/juju/domain/modelmanager/service"
+)
+
+// ControllerConfig provides access to the controller configuration.
+type ControllerConfig interface {
```

Comment:
Ideally, you only expose api sever related api, but I agree the overlap/duplication would be quite high. There is a more practical reason for this, it forces the separation of api server and domain. Allowing db or mock testing.

In reality this might be superfluous as the api facade could have their own interface based on what they need.